### DISCUSSION TITLE
How to better/faster learn/train kakoune/dance shortcuts?

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [71/dance#254] [General]
I have the same problem with vi/vim or any other mode based editor ever since. Do you guys know a way to get traction faster? I still regularly mistype and suddenly replace every letter in a document with the letter 'e' or something odd like that.
or delete text i didn't want to delete.

**Comment by 71** [comment]
Wish I knew myself, I use less than half of the features of Kakoune/Dance and still regularly make mistakes.

With that said, here's how I try to improve: I read about all the commands available in Kakoune (and wrote all the commands available in Dance!) and try to remember them, then when editing code I have these instances where I think "oh, I think a command does that automatically", then go back to the docs to see how it works and what is the relevant keybinding. Over time more of these commands become automatic.

**Reply by sschneider-ihre-pvs** [reply]
Please add Clippy as a reminder for all commands possible :P

**Reply by sschneider-ihre-pvs** [reply]
https://delapouite.github.io/kakoune-tv/

**Reply by sschneider-ihre-pvs** [reply]
I find it really interesting to watch ppl use kakoune, feels weird. but I guess I can learn from watching those scenarios and recall part of them.

**Reply by sschneider-ihre-pvs** [reply]
Do you know why that guy uses
`x`
`a`
instead of `A`?

**Reply by sschneider-ihre-pvs** [reply]
https://delapouite.github.io/kakoune-explain insane

**Reply by 71** [reply]
`x` selects the entire line (**including** the line break), and `a` starts insertion after it (so **after** the line break, ie. at the beginning of the next line).  
`A` starts insertion at the end of the current line (**before** the line break).

**Reply by sschneider-ihre-pvs** [reply]
ahh, now that makes sense
